### PR TITLE
SSAMG interpolation

### PR DESCRIPTION
**Pull request by waynemitchell**
Add unstructured interpolation to SSAMG.

### CODE REVIEW

**Comment by waynemitchell** [comment]
@rfalgout @victorapm , there is still a lot of rough/unfinished code in here for the unstructured interpolation pieces, but we had discussed getting this into recmat sooner rather than later, so I think this is an acceptable state to merge. I've done some testing to make sure I'm not changing behavior when unstructured interpolation is turned off, that is, you should get the same results as the base recmat branch if you don't use the new unstructured interpolation stuff.

File: src/sstruct_ls/ssamg_interp.c [hunk 1]
```diff
@@ -395,37 +396,56 @@ hypre_SSAMGSetupInterpOp( hypre_SStructMatrix  *A,
...
                            {
-                              HYPRE_Real center;
-
-                              center = Pp1[Pi] + Pp2[Pi];
-                              if (center)
-                              {
-                                 Pp1[Pi] /= center;
-                                 Pp2[Pi] /= center;
-                              }
+                              Pp1[Pi] = 0.0;
```
Comment (victorapm) [inline src/sstruct_ls/ssamg_interp.c]: I'm trying to understand the reason for changing the way renormalization is done here

Comment (victorapm) [reply src/sstruct_ls/ssamg_interp.c]: Nevermind, understood!

File: src/sstruct_mv/sstruct_matmult.c [hunk 2]
```diff
@@ -1255,3 +1282,212 @@ hypre_SStructMatrixRTtAP( hypre_SStructMatrix  *RT,
 
    return hypre_error_flag;
 }
+
```
Comment (victorapm) [inline src/sstruct_mv/sstruct_matmult.c]: Should we keep the following methods or move to a separate branch/remove?

File: src/sstruct_mv/sstruct_matrix.c [hunk 3]
```diff
@@ -2173,20 +2135,254 @@ hypre_SStructMatrixToUMatrix( HYPRE_SStructMatrix  matrix,
...
+
+/*--------------------------------------------------------------------------
+ * hypre_SStructMatrixUBoxes WM: todo - is this actually what I want?
+ *
+ * Get a set of boxes that covers points with non-zero rows in the
+ * unstructured matrix.
+ *
+ *--------------------------------------------------------------------------*/
+HYPRE_Int
+hypre_SStructMatrixUBoxes( hypre_SStructMatrix   *A,
```
Comment (victorapm) [inline src/sstruct_mv/sstruct_matrix.c]: Should we move this to other branch and continue its development from there?

**Review by victorapm — APPROVED** [review]
Thanks, Wayne!

File: src/sstruct_mv/sstruct_matrix.c (hunk 3 above)
Comment (waynemitchell) [reply src/sstruct_mv/sstruct_matrix.c]: Agreed. Removing for now.

File: src/sstruct_mv/sstruct_matmult.c (hunk 2 above)
Comment (waynemitchell) [reply src/sstruct_mv/sstruct_matmult.c]: Agreed. Removing for now.

File: src/sstruct_ls/ssamg_csolver.c [hunk 4]
```diff
@@ -99,8 +99,9 @@ hypre_SSAMGCoarseSolverSetup( void *ssamg_vdata )
       {
          HYPRE_BoomerAMGSetMaxIter(csolver, num_crelax);
       }
-      HYPRE_BoomerAMGSetMaxCoarseSize(csolver, 1000);
-      HYPRE_BoomerAMGSetCycleRelaxType(csolver, 0, 3); /* Coarse solver - Jacobi */
+      /* WM: commenting out the large max coarse size and Jacobi coarse solver... why is this here? */
+      /* HYPRE_BoomerAMGSetMaxCoarseSize(csolver, 1000); */
+      /* HYPRE_BoomerAMGSetCycleRelaxType(csolver, 0, 3); /1* Coarse solver - Jacobi *1/ */
```
Comment (waynemitchell) [inline src/sstruct_ls/ssamg_csolver.c]: Should this code be removed?

Comment (victorapm) [reply src/sstruct_ls/ssamg_csolver.c]: Yes, that's a good idea

**Review by victorapm — APPROVED** [review]

**Pull request merged**